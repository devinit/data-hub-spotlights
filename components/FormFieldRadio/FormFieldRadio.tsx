import React, { FunctionComponent } from 'react';

const FormFieldRadio: FunctionComponent = ({ children }) => <div className="form-field__radio-buttons">{children}</div>;

export { FormFieldRadio };
