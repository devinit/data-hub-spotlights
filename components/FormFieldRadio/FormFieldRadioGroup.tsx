import React, { FunctionComponent } from 'react';

interface ComponentProps {
  label: string;
  value: string | string[] | number;
  name: string;
  checked?: boolean;
  onClick?: (value?: string | string[] | number) => void;
}

const FormFieldRadioGroup: FunctionComponent<ComponentProps> = ({ label, name, value, checked, ...props }) => {
  const onClick = (): void => {
    if (props.onClick) {
      props.onClick(value);
    }
  };

  return (
    <div className="form-radio-group cp-form-radio-group-share" onClick={onClick}>
      <input
        className="form-radio"
        type="radio"
        name={name}
        defaultValue={value}
        checked={checked}
        onChange={onClick}
      />
      <label className="form-label" htmlFor={name}>
        {label}
      </label>
    </div>
  );
};

export { FormFieldRadioGroup };
