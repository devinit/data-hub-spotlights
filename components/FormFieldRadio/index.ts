export * from './FormFieldRadio';
export * from './FormFieldRadioGroup';
