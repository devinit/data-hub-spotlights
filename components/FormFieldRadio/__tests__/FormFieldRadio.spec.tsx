/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { FormFieldRadio } from '../FormFieldRadio';

describe('FormFieldRadio', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<FormFieldRadio />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
