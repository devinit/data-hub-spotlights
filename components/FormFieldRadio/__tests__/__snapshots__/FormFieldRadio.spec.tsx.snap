// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldRadio renders correctly 1`] = `
<div
  className="form-field__radio-buttons"
/>
`;
